src/app/modules/time-entries/pages/time-entries.component.ts

```diff
@@ -216,4 +218,18 @@ export class TimeEntriesComponent implements OnInit, OnDestroy {
       });
     }
   }
+
+  checkIfDataSourceIsLoading(data: DataSource<Entry>): boolean {
+    if (data.isLoading) {
+      return true;
+    }
+    return false;
```

Comment:
Good Job Edgar,  you can improve this as follows:
```suggestion
   return data.isLoading;
```
However I also consider eliminating this method because it is irrelevant to return a property, but my recommendation differs from gustavo's, so you can keep the method, but improve it
**Note:** Take care if for some reason data is undefined or null, you can get the following error at execution time:
![image](https://user-images.githubusercontent.com/56373098/130130645-b2543327-90d4-41a2-8691-1f0ec64f3f39.png)

But if you are really sure that it will never be null or undefined, you can leave it as is

Reply:
Thank you for the revision, I am going to make changes to improve my code.